### PR TITLE
feat: add dropdown tree component [KMAPS-423]

### PR DESCRIPTION
**Pull request by nelsonmestevao**
| | |
|:-:|:-:|
| image | image |
| image | image |

### CODE REVIEW

File: packages/ui-components/src/components/tree-item/tree-item.base.scss [hunk 1]
```diff
@@ -8,6 +8,7 @@
 	 * @prop --children-offset: Offset of the child nodes in px.
 
 	 * @prop --connector-lines-color: Line color of children's connectors.
+	 * @prop --vertical-lines-height: Line height of children's connectors.
```
Comment (djcouto) [inline packages/ui-components/src/components/tree-item/tree-item.base.scss:11]: I think it's missing the documentation for the other added CSS props. Not sure if it was intended.

File: packages/ui-components/src/components/tree/tree.base.scss [hunk 2]
```diff
@@ -2,22 +2,28 @@
 
 :host {
 	/**
-	 * @prop --node-width: Tree Node height.
-	 * @prop --node-height: Tree Node height.
-	 * @prop --node-gap: Gap between child nodes in px.
-	 * @prop --children-offset: Offset of the child nodes in px.
-	 *
-	*/
+     * @prop --tree-node-width: Tree Node height.
```
Comment (djcouto) [inline packages/ui-components/src/components/tree/tree.base.scss:5]: Align this please 🙏

File: packages/ui-components/src/components/tree/tree.base.scss [hunk 3]
```diff
@@ -2,22 +2,28 @@
...
+     * @prop --tree-node-height: Tree Node height.
+     * @prop --tree-node-gap: Gap between child nodes in px.
+     * @prop --tree-children-offset: Offset of the child nodes in px.
```
Comment (djcouto) [inline packages/ui-components/src/components/tree/tree.base.scss:8]: Missing the `--tree-children-padding-left` documentation

File: packages/ui-components/src/components/tree/tree.base.scss [hunk 4]
```diff
@@ -2,22 +2,28 @@
...
+     *
+    */
 	--tree-node-width: 300px;
 	--tree-node-height: 40px;
 	--tree-node-gap: #{$spacing-2x};
 	--tree-children-offset: #{$spacing-3x};
+	--tree-children-padding-left: #{$spacing-2x};
+
+	div {
```
Comment (djcouto) [inline packages/ui-components/src/components/tree/tree.base.scss:17]: give this div a meaningful class name and use it here

**Review by djcouto — CHANGES_REQUESTED** [review]
I only left a few comments.

Please add a screenshot to give some context on what you achieved.

Nicely done 👍

**Review by djcouto — APPROVED** [review]

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]
Please have a look why your changes break the `kv-tree`
image

**Comment by nelsonmestevao** [comment]
@ricardogoncalves89 it is fixed now :pray:

**Review by GoncaloCamaz — APPROVED** [review]
Good job! 👍

File: packages/ui-components/src/components/tree-dropdown/tree-dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	@Event() nodeToggleExpand: EventEmitter<ITreeNodeItem>;
+	/** Emitted when the tree node item is clicked */
+	@Event() nodeClick: EventEmitter<ITreeNodeItem>;
+
+	private getHighlightedNodes(node: ITreeNodeItem) {
+		return node.children?.length >= 0 ? { [node.id]: true } : {};
+	}
+
+	render() {
+		console.log(this.nodes);
```
Comment (pedrorebelocosta) [inline packages/ui-components/src/components/tree-dropdown/tree-dropdown.tsx:38]: remove console log

**Review by pedrorebelocosta — CHANGES_REQUESTED** [review]

**Review by ricardogoncalves89 — APPROVED** [review]
🚀

**Review by pedrorebelocosta — APPROVED** [review]

**Pull request merged**